src/pages/en/guides/deploy/github.md

```diff
@@ -25,45 +25,52 @@ You can deploy an Astro site to GitHub Pages by using [GitHub Actions](https://g
     name: Github Pages Astro CI
 
     on:
-      # Trigger the workflow every time you push to the `main` branch
-      # Using a different branch name? Replace `main` with your branch’s name
+     # Trigger the workflow every time you push to the `main` branch
       push:
-        branches: [main]
+        branches: [ main ]
       # Allows you to run this workflow manually from the Actions tab on GitHub.
       workflow_dispatch:
+      
+      # Allow this job to push changes to your repository
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
 
     jobs:
-      deploy:
-        runs-on: ubuntu-20.04
-
-        # Allow this job to push changes to your repository
-        permissions:
-          contents: write
-
+      build:
+        runs-on: ubuntu-latest
         steps:
-          - name: Check out your repository using git
-            uses: actions/checkout@v2
-
-          - name: Use Node.js 16
-            uses: actions/setup-node@v2
-            with:
-              node-version: 16
-
-          # Not using npm? Change `npm ci` to `yarn install` or `pnpm i`
-          - name: Install dependencies
-            run: npm ci
-
-          # Not using npm? Change `npm run build` to `yarn build` or `pnpm run build`
-          - name: Build Astro
-            run: npm run build
+        - uses: actions/checkout@v2
+        - uses: actions/setup-node@v2
+          with:
+            node-version: '16'
+            cache: 'npm'
+        # Not using npm? Change `npm ci` to `yarn install` or `pnpm i`
+        - run: npm ci
+        # Not using npm? Change `npm run build` to `yarn build` or `pnpm run build`
+        - run: npm run build --if-present
```

Comment:
What do you think about including a `name` for each of these steps?

It is more verbose, but I kind of like how it documents what each thing does — especially for useful for those less familiar with GitHub Actions.

```suggestion
        - name: Check out your repository using git
          uses: actions/checkout@v2

        - name: Use Node.js 16
          uses: actions/setup-node@v2
          with:
            node-version: '16'
            cache: 'npm'

        # Not using npm? Change `npm ci` to `yarn install` or `pnpm i`
        - name: Install dependencies
          run: npm ci

        # Not using npm? Change `npm run build` to `yarn build` or `pnpm run build`
        - name: Build Astro
          run: npm run build --if-present
```

Reply:
yeah I like that. Very clear for beginners!